spec/controllers/api/v1/get_an_identity/webhook_messages_controller_spec.rb

```diff
@@ -0,0 +1,146 @@
+          expect(webhook.status).to eq "pending"
+          expect(webhook.processed_at).to be_nil
+          expect(webhook.error).to be_nil
+        end
+
+        it "enqueues a job to process the webhook" do
+          expect {
+            send_request
+          }.to have_enqueued_job(::GetAnIdentity::ProcessWebhookMessageJob)
+        end
+
+        context "when a message is received without the required fields" do
+          let(:raw_message) { "{}" }
+
+          it "returns a 200" do
+            send_request
+            expect(response).to have_http_status(:ok)
+          end
+
+          it "creates a ::GetAnIdentity::WebhookMessage record with thesend_request body parsed into the record" do
+            expect {
+              send_request
+            }.to change(::GetAnIdentity::WebhookMessage, :count).by(1)
+
+            webhook = ::GetAnIdentity::WebhookMessage.last
+            expect(webhook.raw).to eq(message)
+            expect(webhook.message).to be_nil
+            expect(webhook.message_id).to be_nil
+            expect(webhook.message_type).to be_nil
+            expect(webhook.sent_at).to be_nil
+            expect(webhook.status).to eq "pending"
+            expect(webhook.processed_at).to be_nil
+            expect(webhook.error).to be_nil
+          end
+
+          it "enqueues a job to process the webhook" do
+            expect {
+              send_request
+            }.to have_enqueued_job(::GetAnIdentity::ProcessWebhookMessageJob)
+          end
+        end
+
+        context "when received message wasn't sent with the correct signature" do
+          let(:signature_secret) { SecureRandom.uuid }
+
+          it "returns a 401" do
+            send_request
+            expect(response).to have_http_status(:unauthorized)
+          end
+
+          it "does not persist a record" do
+            expect {
+              send_request
+            }.not_to change(::GetAnIdentity::WebhookMessage, :count)
+          end
+        end
+
+        context "when an already received message is repeated" do
+          before do
+            ::GetAnIdentity::WebhookMessage.create!(
+              message: message["message"],
+              message_id: message["notificationId"],
+              message_type: message["messageType"],
+              raw: message,
+              sent_at: Time.zone.parse(message["timeUtc"]),
+            )
+          end
+
+          it "does not create a new webhook record" do
+            expect {
+              send_request
+            }.not_to change(::GetAnIdentity::WebhookMessage, :count)
+          end
+
+          it "does not enqueue a job to process the webhook" do
+            expect {
+              send_request
+            }.not_to have_enqueued_job(::GetAnIdentity::ProcessWebhookMessageJob)
+          end
+        end
+
+        context "when an already received message is repeated but the event type is different" do
```

Comment:
Unless im understanding wrong but should this be:

`'when a duplicate message_id is received but event type is different'` Feels like using the phrase `already received message doesnt quite make sense.`